lib/workers/branch/get-updated.ts

```diff
@@ -98,7 +98,7 @@ export async function getUpdatedPackageFiles(
       }
       if (
         newContent === existingContent &&
-        upgrade.datasource === DATASOURCE_GIT_SUBMODULES
+        upgrade.datasource === datasourceGitSubmodules.id
```

Comment:
@JamieMagee the git submodules-specific code in workers/branch and workers/pr really stands out now. We should think about how we can refactor it sometime

Reply:
@JamieMagee What about adding the commit id as comment to the `.gitmodules` file? So we have a file change

Reply:
Agreed that it sticks out, and should be refactored. Not 100% sure about adding the commit id, but I will create an issue and assign it to myself to track this.